proposals/improved-interpolated-strings.md

```diff
@@ -287,12 +272,106 @@ Do we want to have builders for `System.IFormattable` and `System.FormattableStr
 
 _Answer_: No.
 
-### Lowering
+### Builder pattern codegen
+
+#### `Create` method overload resolution
+
+Given an _applicable\_interpolated\_string\_builder\_type_ `T` and an _interpolated\_string\_expression_ `i`, overload resolution and validation for a valid `Create` method on `T`
+is performed as follows:
+
+1. Member lookup for members with the name `Create` is performed on `T`. The resulting method group is called `M`.
+2. The argument list `A` is constructed as follows:
+    1. The first two arguments are integer constants, representing the literal length of the `i`, and the number of _interpolation_ components in the `i`, respectively.
+    2. If `i` is used as an argument to some parameter `pi` in method `M1`, and parameter `pi` is attributed with `System.Runtime.CompilerServices.InterpolatedStringBuilderArgumentAttribute`,
+    then for every name `Argx` in the `Arguments` array of that attribute the compiler matches it to a parameter `px` that has the same name. The empty string is matched to the receiver
+    of `M1`.
+        * If any `Argx` is not able to be matched to a parameter of `M1`, or an `Argx` requests the receiver of `M1` and `M1` is a static method, an error is produced and no further
+        steps are taken.
+        * Otherwise, the type of every resolved `px` is added to the argument list, in the order specified by the `Arguments` array. These are all passed by value.
+    3. The final argument is a `bool`, passed as an `out` parameter.
+3. Traditional method invocation resolution is performed with method group `M` and argument list `A`.
+    * If a single-best method `F` was found and errors were reported during final validation, those errors are reported and no further steps are taken.
+    * If a single-best method `F` was found and no errors were reported during final validation, the result of overload resolution is `F`.
```

Comment:
I think that this bullet point should be merged with the previous one. If we really want to say something about final validation, which I prefer we don't, we could say that final validation might still produce errors.

Reply:
If the "final validation" mentioned here is the one specified as step 4, then it appears that we are referring to steps out of order

Reply:
> I think that this bullet point should be merged with the previous one. If we really want to say something about final validation, which I prefer we don't, we could say that final validation might still produce errors.

I was trying to word it in the same manner that I plan on implementing it, which would likely suppress cascaded diagnostics if the single-best method failed, say, constraint checking.

> If the "final validation" mentioned here is the one specified as step 4, then it appears that we are referring to steps out of order

It's not the same, I'll reword.